java/src/org/openqa/selenium/grid/sessionqueue/local/LocalNewSessionQueue.java

```diff
@@ -388,14 +385,17 @@ public void complete(
       Lock readLock = lock.readLock();
       readLock.lock();
       Data data;
+      boolean isSessionTimedOut = false;
       try {
         data = requests.get(reqId);
       } finally {
         readLock.unlock();
       }
 
       if (data == null) {
-        return;
+        return false;
+      } else {
+        isSessionTimedOut = isTimedOut(Instant.now(), data);
```

Comment:
The regular timeout handling should already have rejected the session so it is not in the `requests` any more.
So we will never hit this check and it can be removed.

Reply:
Same comment as above, regular timeout handling may not have run here (every 10 secs by default)